HeavyFlavorAnalysis/SpecificDecay/src/BPHDecayToResTrkTrkSameMassBuilderBase.cc

```diff
@@ -0,0 +1,107 @@
+/*
+ *  See header file for a description of this class.
+ *
+ *  \author Paolo Ronchese INFN Padova
+ *
+ */
+
+//-----------------------
+// This Class' Header --
+//-----------------------
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHDecayToResTrkTrkSameMassBuilderBase.h"
+
+//-------------------------------
+// Collaborating Class Headers --
+//-------------------------------
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHDecayGenericBuilderBase.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHDecayToChargedXXbarBuilder.h"
+#include "HeavyFlavorAnalysis/SpecificDecay/interface/BPHParticleNeutralVeto.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHRecoBuilder.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHPlusMinusCandidate.h"
+#include "HeavyFlavorAnalysis/RecoDecay/interface/BPHRecoCandidate.h"
+
+//---------------
+// C++ Headers --
+//---------------
+using namespace std;
+
+//-------------------
+// Initializations --
+//-------------------
+
+//----------------
+// Constructors --
+//----------------
+BPHDecayToResTrkTrkSameMassBuilderBase::BPHDecayToResTrkTrkSameMassBuilderBase(
+    const BPHEventSetupWrapper& es,
+    const string& resName,
+    double resMass,
+    double resWidth,
+    const string& posName,
+    const std::string& negName,
+    double trkMass,
+    double trkSigma,
+    const BPHRecoBuilder::BPHGenericCollection* posCollection,
+    const BPHRecoBuilder::BPHGenericCollection* negCollection)
+    : BPHDecayGenericBuilderBase(es, nullptr),
+      BPHDecayConstrainedBuilderBase(resName, resMass, resWidth),
+      pName(posName),
+      nName(negName),
+      tMass(trkMass),
+      tSigma(trkSigma),
+      pCollection(posCollection),
+      nCollection(negCollection),
+      ptMin(0.0),
+      etaMax(100.0) {}
+
+BPHDecayToResTrkTrkSameMassBuilderBase::BPHDecayToResTrkTrkSameMassBuilderBase(
+    const string& posName,
+    const std::string& negName,
+    double trkMass,
+    double trkSigma,
+    const BPHRecoBuilder::BPHGenericCollection* posCollection,
+    const BPHRecoBuilder::BPHGenericCollection* negCollection)
+    : pName(posName),
+      nName(negName),
+      tMass(trkMass),
+      tSigma(trkSigma),
+      pCollection(posCollection),
+      nCollection(negCollection),
+      ptMin(0.0),
+      etaMax(100.0) {}
+
+//--------------
+// Operations --
+//--------------
+void BPHDecayToResTrkTrkSameMassBuilderBase::fillTrkTrkList() {
+  double mTotMax = massSel->getMassMax();
+  double mTpnMax = mTotMax - (0.8 * rMass);
```

Comment:
This can be moved inside the `if` at L85

Reply:
done